sdk/storage/Azure.Storage.Files.Shares/src/ShareClient.cs

```diff
@@ -346,7 +346,10 @@ protected ShareClient()
                 sasCredential: sasCredential,
                 tokenCredential: tokenCredential,
                 clientDiagnostics: new ClientDiagnostics(options),
-                clientOptions: options);
+                clientOptions: options)
+            {
+                Audience = options.Audience ?? ShareAudience.PublicAudience,
```

Comment:
Ah it's cause I renamed this from `PublicAudience` to `DefaultAudience` cause of archboard feedback. Just change it to that, and this should get resolved.